pbio.go

```diff
@@ -0,0 +1,72 @@
+package pbio
+
+import (
+	"bufio"
+	"encoding/binary"
+	"io"
+
+	"google.golang.org/protobuf/proto"
+)
+
+// Encoder encodes protobuf messages and writes to the underlying writer.
+type Encoder struct {
+	w   io.Writer
+	buf []byte
+}
+
+// NewEncoder inits a new encoder.
+func NewEncoder(w io.Writer) *Encoder {
+	return &Encoder{
+		w:   w,
+		buf: make([]byte, binary.MaxVarintLen64),
+	}
+}
+
+// Encode encodes a message.
+func (e *Encoder) Encode(msg proto.Message) error {
+	sz := proto.Size(msg)
+	e.buf = e.buf[:binary.PutUvarint(e.buf, uint64(sz))]
+
+	data, err := proto.MarshalOptions{}.MarshalAppend(e.buf, msg)
```

Comment:
Maybe you want global proto.MarshalOptions, so you can use it for size as well? `sz := mOpts.Size(msg)`